src/file_resolver.rs

```diff
@@ -86,6 +106,49 @@ impl FileResolver {
         self.files.get(file_no).cloned()
     }
 
+    /// Get the ResolvedFile associated with the `file_no`th `File`
+    pub fn get_resolved_file(&self, file_no: usize) -> Option<ResolvedFile> {
+        self.resolved_files.get(file_no).cloned()
+    }
+
+    /// Get the Solidity name of an `ast::File`. The Solidity name is the name
+    /// relative to the import path of this file.
+    pub fn get_solidity_name(&self, file: &ast::File) -> Option<String> {
+        let cache_no = match file.cache_no {
+            Some(cache_no) => cache_no,
+            _ => {
+                return None;
+            }
+        };
+
+        let import_no = if let Some(resolved_file) = self.resolved_files.get(cache_no) {
+            resolved_file.import_no
+        } else {
+            return None;
+        };
+
+        let full_path = &file.path;
+
+        return match self.import_paths.get(import_no) {
+            Some((map, import_path)) => {
+                let rel_path = if let Ok(rel_path) = full_path.strip_prefix(import_path) {
+                    rel_path
+                } else {
+                    panic!(
+                        "File at path {} should be prefixed by import path {}",
+                        full_path.display(),
+                        import_path.display()
+                    );
+                };
+                match map {
+                    Some(map) => Some(format!("{}/{}", map.to_str().unwrap(), rel_path.display())),
+                    None => Some(format!("{}", rel_path.display())),
+                }
+            }
```

Comment:
Here you are reconstructing the solidity name but this does not work in many cases because the path is canonicalized.

- The filesystem used could be case insensitive,
- There could be symbol links or hard links
-  You're also using `Path.display()` and some characters may display differently
- A path could be relative to the working directory
- I notice a lack of test cases

I don't think this scheme is going to work TBH, It might be better to add an `import_no: Option<usize>` to `ast::File`.

Reply:
Alright, all good points, thank you! I'll go ahead and update.

Also Re missing test cases, it was Friday EOD and I wanted to get high-level feedback on approach before spending an extra couple hours on tests (it's kind of a big repo and I didn't see any test cases in obvious places so it was gonna take a bit of searching). Sorry that wasn't clear, should have mentioned that!

Reply:
I've updated to just track `import_no`.